.github/workflows/docker-image.yml

```diff
@@ -3,6 +3,8 @@ name: Docker Image CI
 on:
   push:
     branches: [ "main" ]
+    paths:
+      - environments/java_codegen_monitoring/Makefile
```

Comment:
Can you pull the version number out into a file with just the version in it. It's easy then to read it in from the Makefile. This way changes to the Makefile itself won't trigger a re-publish accidentally.